src/Illuminate/Support/Collection.php

```diff
@@ -939,6 +960,25 @@ public function transform(callable $callback)
     }
 
     /**
+     * Transpose each item in the collection, interchanging the row and column indexes.
+     *
+     * @return static
+     */
+    public function transpose()
+    {
+        $values = $this->values();
+
+        if (! $values->forAll('is_array')) {
+            throw new InvalidArgumentException('Can only transpose multi-dimensional arrays.');
+        }
```

Comment:
This would prevent people from using a multidimensional collection like this:

```
collect([
    collect([1, 2, 3]),
    collect([4, 5, 6]),
    collect([7, 8, 9]),
]);
```

Maybe check if the children are arrays _or_ instances of `self`?